File: Models/DataBase/Payment.cs
using System.ComponentModel.DataAnnotations;

namespace quan_ly_chuoi_cua_hang_tap_hoa_api.Models.DataBase
{
    public class Payment
    {
        [Key]
        public int PaymentId { get; set; }
        public int InvoiceId { get; set; }
        public decimal Amount { get; set; }
        public string PaymentMethod { get; set; }
        public DateTime PaymentDate { get; set; }

        public virtual Invoice Invoice { get; set; }
    }
}

File: lib/models/payment.dart
class Payment {
  final int paymentId;
  final int invoiceId;
  final double amount;
  final String paymentMethod;
  final DateTime paymentDate;

  Payment({
    required this.paymentId,
    required this.invoiceId,
    required this.amount,
    required this.paymentMethod,
    required this.paymentDate,
  });

  factory Payment.fromJson(Map<String, dynamic> json) => Payment(
        paymentId: json['paymentId'] ?? 0,
        invoiceId: json['invoiceId'] ?? 0,
        amount: (json['amount'] ?? 0).toDouble(),
        paymentMethod: json['paymentMethod'] ?? '',
        paymentDate:
            DateTime.tryParse(json['paymentDate'] ?? '') ?? DateTime.now(),
      );
}
